src/field_theory/fixed.lean

```diff
@@ -159,3 +163,110 @@ by { rw [← fixed_points.minpoly.minimal_polynomial,
+end
+
+variables [fintype G]
+
+lemma fixed_points.dim_le_card : vector_space.dim (fixed_points G F) F ≤ fintype.card G :=
+begin
+  refine dim_le (λ s hs, cardinal.nat_cast_le.1 _),
+  rw [← @dim_fun' F G, ← cardinal.lift_nat_cast.{v (max u v)},
+    cardinal.finset_card, ← cardinal.lift_id (vector_space.dim F (G → F))],
+  exact linear_independent_le_dim'.{_ _ _ (max u v)}
+    (linear_independent_smul_of_linear_independent G F hs)
+end
+
+instance finite_dimensional_fixed_points : finite_dimensional (fixed_points G F) F :=
+finite_dimensional.finite_dimensional_iff_dim_lt_omega.2 $
+lt_of_le_of_lt (fixed_points.dim_le_card G F) (cardinal.nat_lt_omega _)
+
+lemma findim_fixed_points : findim (fixed_points G F) F ≤ fintype.card G :=
+by exact_mod_cast trans_rel_right (≤) (findim_eq_dim _ _) (fixed_points.dim_le_card G F)
+
+lemma linear_independent_to_linear_map (R : Type u) (A : Type v)
+  [comm_semiring R] [integral_domain A] [algebra R A] :
+  linear_independent A (alg_hom.to_linear_map : (A →ₐ[R] A) → (A →ₗ[R] A)) :=
+have linear_independent A (linear_map.lto_fun R A A ∘ alg_hom.to_linear_map),
+from ((linear_independent_monoid_hom A A).comp
+  (coe : (A →ₐ[R] A) → (A →* A))
+  (λ f g hfg, alg_hom.ext $ monoid_hom.ext_iff.1 hfg) : _),
+linear_independent_of_comp _ this
+
+lemma cardinal_mk_alg_hom (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  cardinal.mk (V →ₐ[K] V) ≤ findim V (V →ₗ[K] V) :=
+cardinal_mk_le_findim_of_linear_independent $ linear_independent_to_linear_map K V
+
+noncomputable instance alg_hom.fintype (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  fintype (V →ₐ[K] V) :=
+classical.choice $ cardinal.lt_omega_iff_fintype.1 $
+lt_of_le_of_lt (cardinal_mk_alg_hom K V) (cardinal.nat_lt_omega _)
+
+lemma findim_alg_hom (K : Type u) (V : Type v)
+  [field K] [field V] [algebra K V] [finite_dimensional K V] :
+  fintype.card (V →ₐ[K] V) ≤ findim V (V →ₗ[K] V) :=
+fintype_card_le_findim_of_linear_independent $ linear_independent_to_linear_map K V
+
+/-- Embedding produced from a faithful action. -/
+def to_alg_hom_over_fixed_points (G : Type u) (F : Type v) [group G] [field F]
+  [faithful_mul_semiring_action G F] : G ↪ (F →ₐ[fixed_points G F] F) :=
+{ to_fun := λ g, { commutes' := λ x, x.2 g,
+    .. mul_semiring_action.to_semiring_hom G F g },
+  inj' := λ g₁ g₂ hg, injective_to_semiring_hom G F $ ring_hom.ext $ λ x, alg_hom.ext_iff.1 hg x, }
+
+@[simp] lemma coe_to_alg_hom_over_fixed_points (G : Type u) (F : Type v) [group G] [field F]
+  [faithful_mul_semiring_action G F] :
+  (to_alg_hom_over_fixed_points G F : G → F →ₐ[fixed_points G F] F) =
+    λ g, { commutes' := λ x, x.2 g,
+      .. mul_semiring_action.to_semiring_hom G F g } :=
+rfl
+
+lemma to_alg_hom_over_fixed_points_apply {G : Type u} {F : Type v} [group G] [field F]
+  [faithful_mul_semiring_action G F] (g : G) (x : F) :
+  to_alg_hom_over_fixed_points G F g x = g • x :=
+rfl
```

Comment:
I think nowadays `simps` will build these lemmas for you (using coercions!).

Reply:
No, it generates:
```lean
to_alg_hom_over_fixed_points_to_fun_to_fun :
  ∀ (G : Type u_1) (F : Type u_2) [_inst_4 : group G] [_inst_5 : field F] [_inst_6 : faithful_mul_semiring_action G F]
  (g : G) (a : F),
    (⇑(to_alg_hom_over_fixed_points G F) g).to_fun a = (mul_semiring_action.to_semiring_hom G F g).to_fun a
```

Reply:
@cipher1024 @fpvandoorn Is this something where `simps` could be improved?

Reply:
This is a bug in `@[simps]`. I think/hope that https://github.com/leanprover-community/mathlib/compare/algebra-hom-simps will fix it.

Reply:
(@jcommelin @kckennylau because it is easy to miss comments after a PR is closed)

Reply:
At least, that commit should fix that the LHS is not a coercion. If `simp` can simplify the RHS to `g • a` then `@[simps {simp_rhs := tt}]` should generate the correct RHS.

Reply:
But does it generate `coe_to_alg_hom_over_fixed_points`?

Reply:
Yes, that should be something like `@[simps to_fun {fully_applied := ff}]`. 
* The `to_fun` indicates the projection(s) you want to apply. Here you tell it to not recursively apply any projections of `alg_hom` (the default behavior would probably be `@[simps to_fun_to_fun]`).
* The `{fully_applied := ff}` tells it to not apply the LHS and RHS to `g`.